web/packages/teleport/src/Welcome/Questionnaire/Questionnaire.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { ButtonPrimary, Card, Text } from 'design';
+
+import Validation, { Validator } from 'shared/components/Validation';
+
+import { QuestionnaireFormFields } from './types';
+import { Company } from './Company';
+import { Role } from './Role';
+import { Resources } from './Resources';
+import { supportedResources } from './constants';
+
+export const Questionnaire = () => {
+  const [validResources, setValidResources] = useState<boolean>(undefined);
```

Comment:
Could this be `false` instead?

Reply:
IMO we could go even further an get rid of that custom validation logic for resources.

In `Resoures` we could have something like 
```typescript
const { valid, message } = useRule(requiredArrayField(value));
```

while `requiredArrayField ` would be:

```typescript
const requiredArrayField = (message: string) => (value: []string) => () => {
  const valid = !value.length;
  return {
    valid,
    message: "Resource is required"
  };
};

```

Thanks to this, `resources` would work as any other form field.

WDYT @michellescripts?

Reply:
Sure!